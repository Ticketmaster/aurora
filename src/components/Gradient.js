import PropTypes from "prop-types";
import styled from "styled-components";

import colors from "../theme/colors";
import { mediumAndUp, largeAndUp } from "../theme/mediaQueries";

const Gradient = styled.span`
  z-index: 1;
  background-image: ${props =>
    `linear-gradient(${props.deg.small}, ${props.from}, ${props.to})`};
  ${mediumAndUp`
    background-image: ${props =>
      `linear-gradient(${props.deg.medium}, ${props.from}, ${props.to})`};
  `} ${largeAndUp`
    background-image: ${props =>
      `linear-gradient(${props.deg.large}, ${props.from}, ${props.to})`};
  `};

  &.gradient--overlay:after {
    z-index: 2;
    height: 100%;
    content: "";
    opacity: 0.4;
    top: 0;
    left: 0;
    right: 0;
    position: absolute;
    background-image: linear-gradient(77deg, rgba(0, 0, 0, 0), #000000);
    ${mediumAndUp`
        background-image: linear-gradient(82deg, rgba(0, 0, 0, 0), #000000);
      `};

    ${largeAndUp`
      background-image: linear-gradient(86deg, rgba(0, 0, 0, 0), #000000);
      `};
  }
`;

Gradient.propTypes = {
  deg: PropTypes.shape({
    small: PropTypes.string.isRequired,
    medium: PropTypes.string.isRequired,
    large: PropTypes.string.isRequired
  }),
  from: PropTypes.string,
  to: PropTypes.string
};

Gradient.defaultProps = {
  deg: {
    small: "76deg",
    medium: "80deg",
    large: "81deg"
  },
  from: colors.defaultGradient.from,
  to: colors.defaultGradient.to
};

export default Gradient;
