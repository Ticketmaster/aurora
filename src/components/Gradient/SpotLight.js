import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import GRADIENT_VARIANTS from "./constants";

const styleVariants = {
  standard: {
    width: "auto",
    preserveAspectRatio: "xMinYMid meet"
  },
  special: {
    width: "100%",
    preserveAspectRatio: "none"
  }
};

const SpotLight = ({ className, variant }) => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
    viewBox="0 0 1440 218"
    preserveAspectRatio={styleVariants[variant].preserveAspectRatio}
    className={className}
  >
    <defs>
      <linearGradient id="a" x1="61.654%" x2="3.355%" y1="100%" y2="0%">
        <stop offset="0%" stopColor="#024691" />
        <stop offset="52.996%" stopColor="#BADBFE" />
        <stop offset="100%" stopColor="#C56BFF" />
      </linearGradient>
      <linearGradient
        id="c"
        x1="74.395%"
        x2=".532%"
        y1="129.212%"
        y2="-28.101%"
      >
        <stop offset="0%" stopOpacity="0" />
        <stop offset="100%" stopOpacity=".4" />
        <stop offset="100%" />
        <stop offset="100%" stopOpacity=".4" />
      </linearGradient>
      <path id="b" d="M0 .748h138.365l1325.043 217.52H413.554z" />
    </defs>
    <g fill="none" fillRule="nonzero" opacity=".102" transform="translate(-20)">
      <use fill="url(#a)" xlinkHref="#b" />
      <use fill="url(#c)" xlinkHref="#b" />
    </g>
  </svg>
);

SpotLight.propTypes = {
  className: PropTypes.string,
  variant: PropTypes.oneOf(Object.values(GRADIENT_VARIANTS))
};

SpotLight.defaultProps = {
  className: "",
  variant: "standard"
};

export default styled(SpotLight)`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  width: ${({ variant }) => styleVariants[variant].width};
  height: 100%;
`;
