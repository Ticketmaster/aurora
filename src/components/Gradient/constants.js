import { themes, colors } from "../../theme";

export const DEFAULT_DEG = {
  small: "76deg",
  medium: "80deg",
  large: "81deg"
};

export const DEFAULT_STOPS = [
  themes.global.primary.base,
  colors.defaultGradient.to
];

export const OVERLAY_DEG_NOT_SPOTLIGHT = {
  small: "77deg",
  medium: "82deg",
  large: "86deg"
};

export const OVERLAY_STOPS_NOT_SPOTLIGHT = ["rgba(0, 0, 0, 0)", "#000000"];

export const SPOTLIGHT_DEG = {
  small: "256deg",
  medium: "260deg",
  large: "262deg"
};

export const SPOTLIGHT_STOPS = [
  "rgb(0, 45, 161)",
  `${themes.global.primary.base} 55%`,
  "rgb(0, 45, 161)"
];

export const OVERLAY_SHADOW_DEG = {
  small: "185deg",
  medium: "185deg",
  large: "185deg"
};
export const OVERLAY_SHADOW_STOPS = [
  "rgba(1, 2, 2, 0) 17%",
  "rgba(0, 0, 90, 0) 55%",
  "rgba(00, 0, 0, 1)"
];

export const OVERLAY_SHADOW_TOP_DEG = "to bottom";
export const OVERLAY_SHADOW_TOP_HEIGHT = "26%";
export const OVERLAY_SHADOW_TOP_STOPS = [
  "rgba(0, 0, 0, 0.6)",
  "rgba(0, 0, 0, 0)"
];
