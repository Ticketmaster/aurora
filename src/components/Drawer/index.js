import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import { Consumer } from "./Context";
import {
  DrawerContainer,
  DrawerContent,
  HeaderContent,
  CloseButton
} from "./Drawer.styles";
import Gradient from "../Gradient";
import Text from "../Text/Base";
import HamburgerIcon from "../Icons/HamburgerIcon";
import omitKeys from "../../utils/omitKeys";

export default class Drawer extends React.Component {
  static propTypes = {
    children: PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.element,
      PropTypes.func
    ]),
    header: PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.func,
      PropTypes.string
    ]),
    withGradient: PropTypes.bool
  };

  static defaultProps = { children: null, header: null, withGradient: true };

  getHeaderContent = props => {
    const { header } = this.props;

    if (typeof header === "function") return header(props);

    if (typeof header === "string") {
      return (
        <Text size="kilo" variant="light" primary weight="semiBold">
          {header}
        </Text>
      );
    }

    return header;
  };

  getHeaderLayout = ({ toggleDrawer, isOpen }) => (
    <HeaderContent>
      <div>
        {this.getHeaderContent({
          toggleDrawer,
          isOpen
        })}
      </div>
      <CloseButton type="button" onClick={toggleDrawer}>
        <HamburgerIcon
          className={classNames({
            hamburger: true,
            "hamburger--opened": isOpen,
            "hamburger--closed": !isOpen
          })}
        />
      </CloseButton>
    </HeaderContent>
  );

  renderChildren = props => {
    const { children } = this.props;
    if (typeof children === "function") return children(props);

    return children;
  };

  renderHeader = ({ toggleDrawer, isOpen, withGradient }) => {
    const headerLayout = this.getHeaderLayout({ toggleDrawer, isOpen });
    return withGradient ? (
      <Gradient
        className="gradient--spotlight"
        style={{ height: "60px", position: "relative" }}
      >
        {headerLayout}
      </Gradient>
    ) : (
      headerLayout
    );
  };

  render() {
    const { className, withGradient, ...rest } = this.props;
    return (
      <Consumer>
        {({ isOpen, toggleDrawer }) => (
          <DrawerContainer
            {...omitKeys(["children", "header"], rest)}
            className={classNames(
              { "drawer__content--open": isOpen },
              className
            )}
          >
            <DrawerContent>
              {this.renderHeader({
                toggleDrawer,
                isOpen,
                withGradient
              })}
              {this.renderChildren({
                toggleDrawer,
                isOpen
              })}
            </DrawerContent>
          </DrawerContainer>
        )}
      </Consumer>
    );
  }
}
