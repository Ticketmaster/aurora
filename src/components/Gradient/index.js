import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import colors from "../../theme/colors";
import { mediumAndUp, largeAndUp } from "../../theme/mediaQueries";
import SpotLight from "./SpotLight";
import Angle from "./Angle";
import StyledImageSeo from "../Image/Seo.styles";

const SPOTLIGHT_STOPS = [
  "rgb(0, 45, 161)",
  `${colors.azure.base} 55%`,
  "rgb(0, 45, 161)"
];

const SPOTLIGHT_DEG = {
  small: "256deg",
  medium: "260deg",
  large: "262deg"
};

const GradientStyles = styled.span`
  z-index: 1;
  width: 100%;

  background-image: ${props =>
    `linear-gradient(${props.deg.small}, ${props.stops.join(",")})`};
  ${mediumAndUp`
    background-image: ${props =>
      `linear-gradient(${props.deg.medium}, ${props.stops.join(",")})`};
  `} ${largeAndUp`
    background-image: ${props =>
      `linear-gradient(${props.deg.large}, ${props.stops.join(",")})`};
  `};

  &.gradient--overlay:after,
  &.gradient--underlay:after {
    height: 100%;
    content: "";
    top: 0;
    left: 0;
    right: 0;
    position: absolute;
  }

  &.gradient--spotlight {
    ${largeAndUp`box-shadow: inset 0 1px 40px 10px rgba(0, 0, 0, 0.15);`};
  }

  &.gradient--underlay.gradient--spotlight:after {
    z-index: -1;
    opacity: 0.8;
    background-image: ${props =>
      `linear-gradient(${props.deg.small}, ${props.stops.join(",")})`};
    ${mediumAndUp`
      background-image: ${props =>
        `linear-gradient(${props.deg.medium}, ${props.stops.join(",")})`};
    `} ${largeAndUp`
      background-image: ${props =>
        `linear-gradient(${props.deg.large}, ${props.stops.join(",")})`};
    `};
  }

  &:not(.gradient--spotlight).gradient--overlay:after {
    z-index: 2;
    opacity: 0.4;
    background-image: linear-gradient(77deg, rgba(0, 0, 0, 0), #000000);
    ${mediumAndUp`
        background-image: linear-gradient(82deg, rgba(0, 0, 0, 0), #000000);
      `};

    ${largeAndUp`
      background-image: linear-gradient(86deg, rgba(0, 0, 0, 0), #000000);
      `};
  }

  &:not(.gradient--spotlight).gradient--underlay:after {
    z-index: -1;
    opacity: 0.8;
    background-image: ${props =>
      `linear-gradient(${props.deg.small}, ${props.stops.join(",")})`};
    ${mediumAndUp`
      background-image: ${props =>
        `linear-gradient(${props.deg.medium}, ${props.stops.join(",")})`};
    `} ${largeAndUp`
      background-image: ${props =>
        `linear-gradient(${props.deg.large}, ${props.stops.join(",")})`};
    `};
  }
`;

const SpotLightWrapper = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  overflow: hidden;
`;

class Gradient extends PureComponent {
  render() {
    const {
      src,
      children,
      className,
      stops,
      deg,
      imageRef,
      backgroundRef,
      ...props
    } = this.props;

    const hasSpotLight = className.includes("gradient--spotlight");
    const gradientStops = hasSpotLight ? SPOTLIGHT_STOPS : stops;
    const gradientDeg = hasSpotLight ? SPOTLIGHT_DEG : deg;
    return (
      <GradientStyles
        {...props}
        stops={gradientStops}
        deg={gradientDeg}
        className={className}
        ref={backgroundRef}
      >
        {src && imageRef && <StyledImageSeo src={src} ref={imageRef} />}
        {hasSpotLight && (
          <SpotLightWrapper>
            <SpotLight />
            <Angle />
          </SpotLightWrapper>
        )}
        {children}
      </GradientStyles>
    );
  }
}

Gradient.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  deg: PropTypes.shape({
    small: PropTypes.string.isRequired,
    medium: PropTypes.string.isRequired,
    large: PropTypes.string.isRequired
  }),
  stops: PropTypes.arrayOf(PropTypes.string),
  imageRef: PropTypes.shape({
    current: PropTypes.object
  }),
  backgroundRef: PropTypes.shape({
    current: PropTypes.object
  }),
  src: PropTypes.string
};

Gradient.defaultProps = {
  children: null,
  className: "",
  deg: {
    small: "76deg",
    medium: "80deg",
    large: "81deg"
  },
  stops: [colors.defaultGradient.from, colors.defaultGradient.to],
  imageRef: { current: null },
  backgroundRef: { current: null },
  src: ""
};

export default Gradient;
