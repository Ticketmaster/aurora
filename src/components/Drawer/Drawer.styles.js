import styled from "styled-components";

import constants from "../../theme/constants";
import spacing from "../../theme/spacing";
import getThemeValue from "../../utils/getThemeValue";

export const DrawerContainer = styled.div`
  background-color: ${getThemeValue("onyx", "base")};
  color: ${getThemeValue("white", "base")};
  height: 100%;
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  visibility: hidden;
  transform: translateY(-100%);
  opacity: 0;
  transition: opacity 300ms ${constants.easing.easeInOutQuad};
  overflow: hidden;
  z-index: -1;

  &.drawer__content--open {
    transform: translateY(0);
    visibility: visible;
    opacity: 1;
  }
`;

export const DrawerContent = styled.div`
  display: flex;
  flex-direction: column;
  position: relative;
  width: 100%;
  height: 100vh;
  overflow-y: scroll;
  -webkit-overflow-scrolling: touch;
  overflow-x: hidden;
`;

export const HeaderContent = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: center;
  height: 60px;
  padding-left: ${spacing.moderate};
  position: relative;
  z-index: 5;
`;

export const CloseButton = styled.button`
  appearance: none;
  cursor: pointer;
  border: 0;
  outline: 0;
  background: transparent;
  display: flex;
  align-items: center;
  height: 60px;
  padding: 0 ${spacing.moderate};
`;

export const ContentWrapper = styled.div`
  &.drawer--open {
    overflow: hidden;
  }
`;
