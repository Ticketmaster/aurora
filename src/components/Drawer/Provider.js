/* global window: true */
/* eslint react/sort-comp: off, react/no-unused-state: off */
import React, { Children } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import { Provider } from "./Context";
import { ContentWrapper } from "./Drawer.styles";
import Portal from "../Portal";

export default class DrawerProvider extends React.Component {
  static propTypes = {
    children: PropTypes.element.isRequired,
    className: PropTypes.string
  };

  static defaultProps = { className: "" };

  contentContainer = React.createRef();

  scrollPosition = 0;

  shouldComponentUpdate(prevProps, prevState) {
    return (
      prevState.isOpen !== this.state.isOpen ||
      prevState.content !== this.state.content ||
      prevProps.children !== this.props.children
    );
  }

  componentDidUpdate() {
    const { isOpen } = this.state;
    /* istanbul ignore next */
    if (
      !isOpen &&
      typeof window !== "undefined" &&
      typeof window.scrollTo === "function"
    ) {
      window.scrollTo(0, this.scrollPosition);
    }
  }

  setContent = content => this.setState(() => ({ content }));

  toggleDrawer = () => {
    this.setState(({ isOpen }) => {
      /* istanbul ignore else */
      if (typeof window !== "undefined" && !isOpen) {
        this.scrollPosition = window.pageYOffset;
      }
      return { isOpen: !isOpen };
    });
  };

  state = {
    isOpen: false,
    toggleDrawer: this.toggleDrawer,
    setContent: this.setContent,
    content: null
  };

  render() {
    const { className, children } = this.props;
    const { isOpen, content } = this.state;
    return (
      <Provider value={this.state}>
        <ContentWrapper
          className={classNames({ "drawer--open": isOpen }, className)}
          innerRef={this.contentContainer}
        >
          {Children.only(children)}
          {content && <Portal>{content}</Portal>}
        </ContentWrapper>
      </Provider>
    );
  }
}
