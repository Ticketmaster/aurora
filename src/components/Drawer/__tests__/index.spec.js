import React from "react";
import renderer from "react-test-renderer";

import Drawer from "../index";
import Gradient from "../../Gradient";

jest.mock("../Drawer.styles", () => ({
  DrawerContainer: "DrawerContainer",
  DrawerContent: "DrawerContent",
  HeaderContent: "HeaderContent",
  CloseButton: "CloseButton"
}));

jest.mock("../../Gradient", () => "Gradient");

describe("Drawer", () => {
  it("renders default drawer", () => {
    expect(renderer.create(<Drawer />).toJSON()).toMatchSnapshot();
  });

  it("renders multiple child elements", () => {
    expect(
      renderer
        .create(
          <Drawer>
            <span>Content</span>
            <span>Content</span>
            <span>Content</span>
          </Drawer>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  it("calls a function as child", () => {
    const children = jest.fn(() => "Content");
    renderer.create(<Drawer>{children}</Drawer>);

    expect(children).toHaveBeenCalled();
  });

  it("renders string header header", () => {
    expect(
      renderer.create(<Drawer header="Some Title">Content</Drawer>).toJSON()
    ).toMatchSnapshot();
  });

  it("renders header elements", () => {
    expect(
      renderer
        .create(
          <Drawer header={<span>Some Title Element</span>}>
            <span>Content</span>
          </Drawer>
        )
        .toJSON()
    ).toMatchSnapshot();
  });

  it("calls header as function", () => {
    const header = jest.fn(() => "Header Content");
    renderer.create(<Drawer header={header}>Content</Drawer>);

    expect(header).toHaveBeenCalled();
  });

  it("renders Header WITH Gradient by default", () => {
    const header = "Header";
    const wrapper = renderer.create(<Drawer header={header}>Content</Drawer>);

    expect(wrapper.root.findAllByType(Gradient)).toHaveLength(1);
  });

  it("renders Header WITH Gradient when the 'withSpotLight' is true", () => {
    const header = "Header";
    const wrapper = renderer.create(
      <Drawer withSpotLight header={header}>
        Content
      </Drawer>
    );

    expect(wrapper.root.findAllByType(Gradient)).toHaveLength(1);
  });

  it("renders Header WITHOUT Gradient when 'withSpotLight' is false", () => {
    const header = "Header";
    const wrapper = renderer.create(
      <Drawer withSpotLight={false} header={header}>
        Content
      </Drawer>
    );

    expect(wrapper.root.findAllByType(Gradient)).toHaveLength(0);
  });
});
