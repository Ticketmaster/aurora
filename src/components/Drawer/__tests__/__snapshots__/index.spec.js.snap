// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer renders default drawer 1`] = `
<div
  className="kDkfKC"
>
  <div
    className="bwUIrh"
  >
    <span
      className="gradient--spotlight oFJbF"
      style={
        Object {
          "height": "60px",
          "position": "relative",
        }
      }
    >
      
      <div
        className="nBCnm"
      >
        <svg
          className="czenYD"
          preserveAspectRatio="xMinYMid meet"
          viewBox="0 0 1440 218"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <linearGradient
              id="a"
              x1="61.654%"
              x2="3.355%"
              y1="100%"
              y2="0%"
            >
              <stop
                offset="0%"
                stopColor="#024691"
              />
              <stop
                offset="52.996%"
                stopColor="#BADBFE"
              />
              <stop
                offset="100%"
                stopColor="#C56BFF"
              />
            </linearGradient>
            <linearGradient
              id="c"
              x1="74.395%"
              x2=".532%"
              y1="129.212%"
              y2="-28.101%"
            >
              <stop
                offset="0%"
                stopOpacity="0"
              />
              <stop
                offset="100%"
                stopOpacity=".4"
              />
              <stop
                offset="100%"
              />
              <stop
                offset="100%"
                stopOpacity=".4"
              />
            </linearGradient>
            <path
              d="M0 .748h138.365l1325.043 217.52H413.554z"
              id="b"
            />
          </defs>
          <g
            fill="none"
            fillRule="nonzero"
            opacity=".102"
            transform="translate(-20)"
          >
            <use
              fill="url(#a)"
              xlinkHref="#b"
            />
            <use
              fill="url(#c)"
              xlinkHref="#b"
            />
          </g>
        </svg>
        <svg
          className="gWfTKb"
          preserveAspectRatio="xMaxYMid meet"
          viewBox="0 0 476 218"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <linearGradient
              id="Angle-linearGradient-1"
              x1="21.6784013%"
              x2="53.474493%"
              y1="112.586732%"
              y2="23.9282924%"
            >
              <stop
                offset="0%"
                stopColor="#000000"
                stopOpacity="0"
              />
              <stop
                offset="100%"
                stopColor="#000000"
              />
              <stop
                offset="100%"
                stopColor="#000000"
              />
            </linearGradient>
          </defs>
          <g
            fill="none"
            fillRule="evenodd"
            opacity="0.0490513393"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="url(#Angle-linearGradient-1)"
              fillRule="nonzero"
              transform="translate(-1016.000000, -184.000000)"
            >
              <g
                transform="translate(52.000000, 184.000000)"
              >
                <g
                  transform="translate(-20.000000, 0.000000)"
                >
                  <polygon
                    points="984.668337 0 1460 218 984.668337 218"
                    transform="translate(1222.334169, 109.000000) scale(-1, 1) translate(-1222.334169, -109.000000) "
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div
        className="fgRSsp"
      >
        <div />
        <button
          className="fKLvED"
          onClick={undefined}
          type="button"
        >
          <span
            className="hamburger hamburger--closed hjDmZH"
          >
            <span
              className="kpXvGB"
            />
          </span>
        </button>
      </div>
    </span>
  </div>
</div>
`;

exports[`Drawer renders header elements 1`] = `
<div
  className="kDkfKC"
>
  <div
    className="bwUIrh"
  >
    <span
      className="gradient--spotlight oFJbF"
      style={
        Object {
          "height": "60px",
          "position": "relative",
        }
      }
    >
      
      <div
        className="nBCnm"
      >
        <svg
          className="czenYD"
          preserveAspectRatio="xMinYMid meet"
          viewBox="0 0 1440 218"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <linearGradient
              id="a"
              x1="61.654%"
              x2="3.355%"
              y1="100%"
              y2="0%"
            >
              <stop
                offset="0%"
                stopColor="#024691"
              />
              <stop
                offset="52.996%"
                stopColor="#BADBFE"
              />
              <stop
                offset="100%"
                stopColor="#C56BFF"
              />
            </linearGradient>
            <linearGradient
              id="c"
              x1="74.395%"
              x2=".532%"
              y1="129.212%"
              y2="-28.101%"
            >
              <stop
                offset="0%"
                stopOpacity="0"
              />
              <stop
                offset="100%"
                stopOpacity=".4"
              />
              <stop
                offset="100%"
              />
              <stop
                offset="100%"
                stopOpacity=".4"
              />
            </linearGradient>
            <path
              d="M0 .748h138.365l1325.043 217.52H413.554z"
              id="b"
            />
          </defs>
          <g
            fill="none"
            fillRule="nonzero"
            opacity=".102"
            transform="translate(-20)"
          >
            <use
              fill="url(#a)"
              xlinkHref="#b"
            />
            <use
              fill="url(#c)"
              xlinkHref="#b"
            />
          </g>
        </svg>
        <svg
          className="gWfTKb"
          preserveAspectRatio="xMaxYMid meet"
          viewBox="0 0 476 218"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <linearGradient
              id="Angle-linearGradient-1"
              x1="21.6784013%"
              x2="53.474493%"
              y1="112.586732%"
              y2="23.9282924%"
            >
              <stop
                offset="0%"
                stopColor="#000000"
                stopOpacity="0"
              />
              <stop
                offset="100%"
                stopColor="#000000"
              />
              <stop
                offset="100%"
                stopColor="#000000"
              />
            </linearGradient>
          </defs>
          <g
            fill="none"
            fillRule="evenodd"
            opacity="0.0490513393"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="url(#Angle-linearGradient-1)"
              fillRule="nonzero"
              transform="translate(-1016.000000, -184.000000)"
            >
              <g
                transform="translate(52.000000, 184.000000)"
              >
                <g
                  transform="translate(-20.000000, 0.000000)"
                >
                  <polygon
                    points="984.668337 0 1460 218 984.668337 218"
                    transform="translate(1222.334169, 109.000000) scale(-1, 1) translate(-1222.334169, -109.000000) "
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div
        className="fgRSsp"
      >
        <div>
          <span>
            Some Title Element
          </span>
        </div>
        <button
          className="fKLvED"
          onClick={undefined}
          type="button"
        >
          <span
            className="hamburger hamburger--closed hjDmZH"
          >
            <span
              className="kpXvGB"
            />
          </span>
        </button>
      </div>
    </span>
    <span>
      Content
    </span>
  </div>
</div>
`;

exports[`Drawer renders multiple child elements 1`] = `
<div
  className="kDkfKC"
>
  <div
    className="bwUIrh"
  >
    <span
      className="gradient--spotlight oFJbF"
      style={
        Object {
          "height": "60px",
          "position": "relative",
        }
      }
    >
      
      <div
        className="nBCnm"
      >
        <svg
          className="czenYD"
          preserveAspectRatio="xMinYMid meet"
          viewBox="0 0 1440 218"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <linearGradient
              id="a"
              x1="61.654%"
              x2="3.355%"
              y1="100%"
              y2="0%"
            >
              <stop
                offset="0%"
                stopColor="#024691"
              />
              <stop
                offset="52.996%"
                stopColor="#BADBFE"
              />
              <stop
                offset="100%"
                stopColor="#C56BFF"
              />
            </linearGradient>
            <linearGradient
              id="c"
              x1="74.395%"
              x2=".532%"
              y1="129.212%"
              y2="-28.101%"
            >
              <stop
                offset="0%"
                stopOpacity="0"
              />
              <stop
                offset="100%"
                stopOpacity=".4"
              />
              <stop
                offset="100%"
              />
              <stop
                offset="100%"
                stopOpacity=".4"
              />
            </linearGradient>
            <path
              d="M0 .748h138.365l1325.043 217.52H413.554z"
              id="b"
            />
          </defs>
          <g
            fill="none"
            fillRule="nonzero"
            opacity=".102"
            transform="translate(-20)"
          >
            <use
              fill="url(#a)"
              xlinkHref="#b"
            />
            <use
              fill="url(#c)"
              xlinkHref="#b"
            />
          </g>
        </svg>
        <svg
          className="gWfTKb"
          preserveAspectRatio="xMaxYMid meet"
          viewBox="0 0 476 218"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <linearGradient
              id="Angle-linearGradient-1"
              x1="21.6784013%"
              x2="53.474493%"
              y1="112.586732%"
              y2="23.9282924%"
            >
              <stop
                offset="0%"
                stopColor="#000000"
                stopOpacity="0"
              />
              <stop
                offset="100%"
                stopColor="#000000"
              />
              <stop
                offset="100%"
                stopColor="#000000"
              />
            </linearGradient>
          </defs>
          <g
            fill="none"
            fillRule="evenodd"
            opacity="0.0490513393"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="url(#Angle-linearGradient-1)"
              fillRule="nonzero"
              transform="translate(-1016.000000, -184.000000)"
            >
              <g
                transform="translate(52.000000, 184.000000)"
              >
                <g
                  transform="translate(-20.000000, 0.000000)"
                >
                  <polygon
                    points="984.668337 0 1460 218 984.668337 218"
                    transform="translate(1222.334169, 109.000000) scale(-1, 1) translate(-1222.334169, -109.000000) "
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div
        className="fgRSsp"
      >
        <div />
        <button
          className="fKLvED"
          onClick={undefined}
          type="button"
        >
          <span
            className="hamburger hamburger--closed hjDmZH"
          >
            <span
              className="kpXvGB"
            />
          </span>
        </button>
      </div>
    </span>
    <span>
      Content
    </span>
    <span>
      Content
    </span>
    <span>
      Content
    </span>
  </div>
</div>
`;

exports[`Drawer renders string header header 1`] = `
<div
  className="kDkfKC"
>
  <div
    className="bwUIrh"
  >
    <span
      className="gradient--spotlight oFJbF"
      style={
        Object {
          "height": "60px",
          "position": "relative",
        }
      }
    >
      
      <div
        className="nBCnm"
      >
        <svg
          className="czenYD"
          preserveAspectRatio="xMinYMid meet"
          viewBox="0 0 1440 218"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <linearGradient
              id="a"
              x1="61.654%"
              x2="3.355%"
              y1="100%"
              y2="0%"
            >
              <stop
                offset="0%"
                stopColor="#024691"
              />
              <stop
                offset="52.996%"
                stopColor="#BADBFE"
              />
              <stop
                offset="100%"
                stopColor="#C56BFF"
              />
            </linearGradient>
            <linearGradient
              id="c"
              x1="74.395%"
              x2=".532%"
              y1="129.212%"
              y2="-28.101%"
            >
              <stop
                offset="0%"
                stopOpacity="0"
              />
              <stop
                offset="100%"
                stopOpacity=".4"
              />
              <stop
                offset="100%"
              />
              <stop
                offset="100%"
                stopOpacity=".4"
              />
            </linearGradient>
            <path
              d="M0 .748h138.365l1325.043 217.52H413.554z"
              id="b"
            />
          </defs>
          <g
            fill="none"
            fillRule="nonzero"
            opacity=".102"
            transform="translate(-20)"
          >
            <use
              fill="url(#a)"
              xlinkHref="#b"
            />
            <use
              fill="url(#c)"
              xlinkHref="#b"
            />
          </g>
        </svg>
        <svg
          className="gWfTKb"
          preserveAspectRatio="xMaxYMid meet"
          viewBox="0 0 476 218"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <linearGradient
              id="Angle-linearGradient-1"
              x1="21.6784013%"
              x2="53.474493%"
              y1="112.586732%"
              y2="23.9282924%"
            >
              <stop
                offset="0%"
                stopColor="#000000"
                stopOpacity="0"
              />
              <stop
                offset="100%"
                stopColor="#000000"
              />
              <stop
                offset="100%"
                stopColor="#000000"
              />
            </linearGradient>
          </defs>
          <g
            fill="none"
            fillRule="evenodd"
            opacity="0.0490513393"
            stroke="none"
            strokeWidth="1"
          >
            <g
              fill="url(#Angle-linearGradient-1)"
              fillRule="nonzero"
              transform="translate(-1016.000000, -184.000000)"
            >
              <g
                transform="translate(52.000000, 184.000000)"
              >
                <g
                  transform="translate(-20.000000, 0.000000)"
                >
                  <polygon
                    points="984.668337 0 1460 218 984.668337 218"
                    transform="translate(1222.334169, 109.000000) scale(-1, 1) translate(-1222.334169, -109.000000) "
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
      </div>
      <div
        className="fgRSsp"
      >
        <div>
          <div
            className="text text--light text--primary bdJGfZ"
            disabled={false}
            size="kilo"
          >
            Some Title
          </div>
        </div>
        <button
          className="fKLvED"
          onClick={undefined}
          type="button"
        >
          <span
            className="hamburger hamburger--closed hjDmZH"
          >
            <span
              className="kpXvGB"
            />
          </span>
        </button>
      </div>
    </span>
    Content
  </div>
</div>
`;
